keel-core/src/main/kotlin/com/netflix/spinnaker/keel/actuation/EnvironmentPromotionChecker.kt

```diff
@@ -22,9 +22,18 @@ class EnvironmentPromotionChecker(
   private val constraints: List<ConstraintEvaluator<*>>,
   private val publisher: ApplicationEventPublisher
 ) {
```

Comment:
Separating out the four different type of constraint evaluators here so that they can be called in a specific order.